app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataValuePresenter.kt

```diff
@@ -0,0 +1,741 @@
+                        view.context.getString(R.string.missing_mandatory_fields_title),
+                        view.context.resources.getString(R.string.field_mandatory)
+                    )
+                } else {
+                    view.showAlertDialog(
+                        view.context.getString(R.string.missing_mandatory_fields_title),
+                        view.context.resources.getString(R.string.field_required)
+                    )
+                }
+            } else {
+                view.isOpenOrReopen
+                disposable.add(
+                    repository.reopenDataSet(orgUnitUid, periodId, attributeOptionCombo)
+                        .subscribeOn(schedulerProvider.io())
+                        .observeOn(schedulerProvider.ui())
+                        .subscribe(
+                            { reopen ->
+                                view.setCompleteReopenText(false)
+                                view.update(reopen!!)
+                            },
+                            { Timber.e(it) }
+                        )
+                )
+            }
+        }
+    }
+
+    private fun checkAllFieldRequired(): Boolean {
+        var allFields = true
+        tableCells.forEach { table ->
+            table.forEach { row ->
+                val fieldsWithValue = dataTableModel?.dataValues()
+                    ?.filter { dataSetTableModel ->
+                        dataSetTableModel.dataElement() == row.first().dataElement()
+                    }
+
+                if (!fieldsWithValue.isNullOrEmpty() && fieldsWithValue.size != row.size) {
+                    allFields = false
+                    view.highligthHeaderRow(
+                        tableCells.indexOf(table),
+                        table.indexOf(row),
+                        false
+                    )
+                }
+            }
+        }
+        return allFields
+    }
+
+    private fun checkMandatoryField(): Boolean {
+        var mandatoryOk = true
+        tableCells.forEach { table ->
+            table.forEach { row ->
+                row.forEach { field ->
+                    val fieldWithValue = dataTableModel?.dataValues()
+                        ?.filter { dataSetTableModel ->
+                            dataSetTableModel.dataElement() == field.dataElement() &&
+                                dataSetTableModel.categoryOptionCombo() ==
+                                field.categoryOptionCombo()
+                        }
+
+                    if (field.editable()!! && field.mandatory() && fieldWithValue.isNullOrEmpty()) {
+                        mandatoryOk = false
+                        view.highligthHeaderRow(
+                            tableCells.indexOf(table),
+                            table.indexOf(row),
+                            true
+                        )
+                    }
+                }
+            }
+        }
+        return mandatoryOk
+    }
+
```

Comment:
These are the modified methods

Reply:
Just so the others reviewers know:

- checkAllFieldRequired
- checkMandatoryField

Reply:
any unit test?